core/src/main/java/com/novoda/landingstrip/Notifier.java

```diff
@@ -0,0 +1,34 @@
+package com.novoda.landingstrip;
+
+import android.view.View;
+
+class Notifier implements LandingStripAdapter.Listener {
+
+    private final TabsContainerView tabsContainerView;
+
+    Notifier(TabsContainerView tabsContainerView) {
+        this.tabsContainerView = tabsContainerView;
+    }
+
+    @Override
+    public <T extends View> void onNotifyDataSetChanged(LandingStripAdapter<T> adapter) {
+        recreateAndBindTabs(adapter);
+    }
+
+    private <T extends View> void recreateAndBindTabs(LandingStripAdapter<T> adapter) {
```

Comment:
Also, in the future we should be [recycling the views](https://github.com/rock3r/squanchy/blob/develop/app/src/main/java/net/squanchy/support/widget/SpeakerView.java#L79) if possible. We might need some sort of contract such as view types in the adapter, but that shouldn't be too hard. Currently this operation can be rather expensive if you have several tabs.